File: src/test/java/pirateboat/multifileHoster/AlldebridTest.java
package pirateboat.multifileHoster;

import java.util.List;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import pirateboat.torrent.Torrent;
import pirateboat.torrent.TorrentFile;
import pirateboat.utilities.HttpHelper;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

class AlldebridTest {

    private Alldebrid alldebrid;

    @BeforeEach
    void beforeEach() {
        alldebrid = new Alldebrid(new HttpHelper());
    }

    @Disabled
    @Test
    void getRemoteTorrents() {
        // Given
        // When
        final List<Torrent> remoteTorrents = alldebrid.getRemoteTorrents();
        // Then
        assertTrue(remoteTorrents != null);
    }

    @Disabled
    @Test
    void getFilesFromTorrent() {
        // Given
        final List<Torrent> remoteTorrents = alldebrid.getRemoteTorrents();
        // When
        final Torrent torrent = remoteTorrents.stream().findFirst().orElse(null);
        if(torrent != null) {
            final List<TorrentFile> filesFromTorrent = alldebrid.getFilesFromTorrent(torrent);
            assertNotNull(filesFromTorrent);
        }
        assertTrue(true);
    }

}

File: src/test/java/boat/multifileHoster/AlldebridTest.kt
package boat.multifileHoster

import boat.torrent.Torrent
import boat.utilities.HttpHelper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertDoesNotThrow
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

@Disabled
internal class AlldebridTest {
    private lateinit var alldebrid: Alldebrid

    @BeforeEach
    fun beforeEach() {
        alldebrid = Alldebrid(HttpHelper())
    }

    @Test
    fun getRemoteTorrents() {
        // Given
        // When
        // Then
        assertDoesNotThrow { alldebrid.getRemoteTorrents() }
    }


    @Test
    fun filesFromTorrent() {
        // Given
        val remoteTorrents: List<Torrent?> = alldebrid.getRemoteTorrents()
        // When
        val torrent = remoteTorrents.stream().findFirst().orElse(null)
        if (torrent != null) {
            val filesFromTorrent = alldebrid.getFilesFromTorrent(torrent)
            Assertions.assertNotNull(filesFromTorrent)
        }
        Assertions.assertTrue(true)
    }

    @Test
    fun remainingTrafficInMB() {
        // Given
        // When
        // Then
        assertDoesNotThrow { alldebrid.getRemainingTrafficInMB() }
    }
}

